# Regression Linéaire Multiple

#  Importation de NumPy pour les calculs numériques et la manipulation de tableaux (arrays)
import numpy as np

# Importation de Pandas pour la manipulation et l'analyse de données structurées (DataFrames)
import pandas as pd

# Importation de Pyplot (de Matplotlib) pour la création de graphiques et de visualisations de base
import matplotlib.pyplot as plt

# Importation de Seaborn pour les visualisations statistiques avancées et esthétiques
import sns

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Chargement du fichier CSV contenant les 3000 données Mpox de la RDC dans un DataFrame Pandas nommé 'dflm'
dflm = pd.read_csv("E:/Documents/COURS M2_MASG/data_Science_IA/donnees_mpox_RDC_3000.csv")

# Affichage des premières et dernières lignes du DataFrame 'dflm' pour inspecter visuellement la structure des données
dflm

In [3]:
dflm = pd.read_csv("E:/Documents/COURS M2_MASG/data_Science_IA/donnees_mpox_RDC_3000.csv")
dflm

,ID,Semaine,Province,Pluviometrie_mm,Temperature_C,NDVI,Humidite_pct,Densite_Population,Couverture_Vaccinale_pct,Tests_Realises,Distance_Centre_Sante_km,Reservoirs_Animaux,Mobilite_Humaine,Cas_Confirmes,Saison,Population_Risque,Taux_Positivite_pct
0,OBS_00001,1,Mongala,206,24.4,0.480,64,39,31.8,150,22,1,4,29,Pluie,3478,19.33
1,OBS_00002,2,Kinshasa,56,23.7,0.474,74,107,24.9,300,5,1,6,42,Seche,3713,14.00
2,OBS_00003,3,Kinshasa,59,25.2,0.333,62,140,53.9,310,16,0,8,20,Seche,4833,6.45
3,OBS_00004,4,Mai-Ndombe,396,24.1,0.354,68,97,25.9,210,4,0,10,82,Pluie,3777,39.05
4,OBS_00005,5,Kasaï,127,19.8,0.303,61,86,30.3,260,24,1,6,24,Pluie,9396,9.23
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2995,OBS_02996,2996,Kongo Central,271,31.6,0.534,88,57,22.4,220,33,1,7,125,Pluie,3166,56.82
2996,OBS_02997,2997,Sud-Ubangi,88,24.6,0.262,95,47,34.1,370,7,1,6,70,Seche,6099,18.92
2997,OBS_02998,2998,Bas-Uele,268,25.6,0.720,58,18,41.9,190,23,0,2,36,Pluie,5475,18.95
2998,OBS_02999,2999,Haut-Uele,211,24.9,0.196,77,28,10.6,370,2,0,2,160,Pluie,4813,43.24


# Génère un résumé statistique descriptif des variables numériques du DataFrame (compte, moyenne, écart-type, valeurs minimales/maximales et quartiles)
dflm.describe()

In [4]:
dflm.describe()

,Semaine,Pluviometrie_mm,Temperature_C,NDVI,Humidite_pct,Densite_Population,Couverture_Vaccinale_pct,Tests_Realises,Distance_Centre_Sante_km,Reservoirs_Animaux,Mobilite_Humaine,Cas_Confirmes,Population_Risque,Taux_Positivite_pct
count,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.00000,3000.000000,3000.000000,3000.000000,3000.000000
mean,1500.500000,157.779667,24.980367,0.451182,71.571667,119.100333,31.843967,286.143333,15.558333,0.40000,5.018000,79.603333,5471.297667,27.041943
std,866.169729,101.958774,3.832510,0.157799,11.625285,95.218460,14.079517,88.256194,15.139386,0.48998,1.954213,70.119799,2636.188129,20.156959
min,1.000000,2.000000,18.000000,0.103000,40.000000,5.000000,5.600000,90.000000,1.000000,0.00000,1.000000,0.000000,1001.000000,0.000000
25%,750.750000,77.000000,22.300000,0.331000,64.000000,48.000000,20.800000,220.000000,5.000000,0.00000,4.000000,29.000000,3127.000000,11.430000
50%,1500.500000,136.000000,24.900000,0.452000,71.500000,94.000000,30.400000,280.000000,11.000000,0.00000,5.000000,57.000000,5514.500000,21.455000
75%,2250.250000,221.000000,27.600000,0.576000,80.000000,163.000000,41.525000,340.000000,22.000000,1.00000,6.000000,109.000000,7819.250000,38.070000
max,3000.000000,400.000000,34.000000,0.795000,95.000000,500.000000,77.800000,800.000000,142.000000,1.00000,10.000000,479.000000,9997.000000,95.000000


# Calcule le nombre de valeurs manquantes (NaN) pour chaque colonne du DataFrame en combinant isnull() et sum()
count = dflm.isnull().sum()

# Affiche à l'écran le décompte des valeurs manquantes obtenu pour chaque variable
print(count)

In [5]:
count = dflm.isnull().sum()
print(count)

ID                          0
Semaine                     0
Province                    0
Pluviometrie_mm             0
Temperature_C               0
NDVI                        0
Humidite_pct                0
Densite_Population          0
Couverture_Vaccinale_pct    0
Tests_Realises              0
Distance_Centre_Sante_km    0
Reservoirs_Animaux          0
Mobilite_Humaine            0
Cas_Confirmes               0
Saison                      0
Population_Risque           0
Taux_Positivite_pct         0
dtype: int64


# Calcule le pourcentage de valeurs manquantes pour chaque colonne en divisant le nombre de valeurs manquantes par le nombre total de lignes (len(dflm)), puis en multipliant par 100
missing_data = count/len(dflm)*100

# Affiche à l'écran le pourcentage de données manquantes obtenu pour chaque variable afin d'évaluer l'impact sur l'analyse
print(missing_data)

In [6]:
missing_data = count/len(dflm)*100
print(missing_data)

ID                          0.0
Semaine                     0.0
Province                    0.0
Pluviometrie_mm             0.0
Temperature_C               0.0
NDVI                        0.0
Humidite_pct                0.0
Densite_Population          0.0
Couverture_Vaccinale_pct    0.0
Tests_Realises              0.0
Distance_Centre_Sante_km    0.0
Reservoirs_Animaux          0.0
Mobilite_Humaine            0.0
Cas_Confirmes               0.0
Saison                      0.0
Population_Risque           0.0
Taux_Positivite_pct         0.0
dtype: float64


# Supprime toutes les lignes du DataFrame 'dflm' qui contiennent au moins une valeur manquante (NaN) pour nettoyer la base de données
dflm = dflm.dropna()

# Affiche les dimensions actuelles du DataFrame sous forme de tuple (nombre de lignes, nombre de colonnes) après la suppression des valeurs manquantes
dflm.shape

In [7]:
dflm = dflm.dropna()
dflm.shape

(3000, 17)

# Affiche un résumé technique complet du DataFrame : le type d'objet, le nombre total de lignes (entrées), les noms des colonnes avec leur nombre de valeurs non nulles, et le type de données (Dtype) de chaque variable
dflm.info()

In [8]:
dflm.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3000 entries, 0 to 2999
Data columns (total 17 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   ID                        3000 non-null   object 
 1   Semaine                   3000 non-null   int64  
 2   Province                  3000 non-null   object 
 3   Pluviometrie_mm           3000 non-null   int64  
 4   Temperature_C             3000 non-null   float64
 5   NDVI                      3000 non-null   float64
 6   Humidite_pct              3000 non-null   int64  
 7   Densite_Population        3000 non-null   int64  
 8   Couverture_Vaccinale_pct  3000 non-null   float64
 9   Tests_Realises            3000 non-null   int64  
 10  Distance_Centre_Sante_km  3000 non-null   int64  
 11  Reservoirs_Animaux        3000 non-null   int64  
 12  Mobilite_Humaine          3000 non-null   int64  
 13  Cas_Confirmes             3000 non-null   int64  
 14  Saison  

# Renvoie et affiche la liste contenant les noms de toutes les colonnes (variables) présentes dans le DataFrame 'dflm'
dflm.columns

In [9]:
dflm.columns

Index(['ID', 'Semaine', 'Province', 'Pluviometrie_mm', 'Temperature_C', 'NDVI',
       'Humidite_pct', 'Densite_Population', 'Couverture_Vaccinale_pct',
       'Tests_Realises', 'Distance_Centre_Sante_km', 'Reservoirs_Animaux',
       'Mobilite_Humaine', 'Cas_Confirmes', 'Saison', 'Population_Risque',
       'Taux_Positivite_pct'],
      dtype='object')

# Crée un nouveau DataFrame 'df_txt' en filtrant et en sélectionnant uniquement les colonnes de type "object" (les variables textuelles ou qualitatives) du DataFrame initial 'dflm'
df_txt = dflm.select_dtypes(include=["object"])

# Affichage des premières et dernières lignes du nouveau DataFrame 'df_txt' pour inspecter les variables qualitatives de la base Mpox
df_txt

In [10]:
df_txt = dflm.select_dtypes(include=["object"])
df_txt

,ID,Province,Saison
0,OBS_00001,Mongala,Pluie
1,OBS_00002,Kinshasa,Seche
2,OBS_00003,Kinshasa,Seche
3,OBS_00004,Mai-Ndombe,Pluie
4,OBS_00005,Kasaï,Pluie
...,...,...,...
2995,OBS_02996,Kongo Central,Pluie
2996,OBS_02997,Sud-Ubangi,Seche
2997,OBS_02998,Bas-Uele,Pluie
2998,OBS_02999,Haut-Uele,Pluie


# Renvoie et affiche la liste des noms de toutes les colonnes contenant des variables textuelles ou qualitatives (de type "object")
df_txt.columns

In [ ]:
df_txt.columns

# Importation de la classe LabelEncoder de la bibliothèque scikit-learn pour encoder les variables catégorielles en valeurs numériques
from sklearn.preprocessing import LabelEncoder

# Initialisation d'une instance de l'encodeur nommée 'le'
le = LabelEncoder()

# Définition de la liste des colonnes textuelles à transformer (la province, la saison et l'identifiant)
columns = ['Province', 'Saison', 'ID']

# Boucle 'for' qui parcourt une à une les colonnes spécifiées dans la liste 'columns'
for i in columns :
    # Applique l'encodage : convertit les modalités textuelles de la colonne 'i' en entiers (0, 1, 2, etc.) et remplace l'ancienne colonne par ces nouvelles valeurs numériques
    dflm[i] = le.fit_transform(dflm[i])

In [12]:
from sklearn.preprocessing import LabelEncoder

le = LabelEncoder()

columns = ['Province','Saison','ID']
for i in columns :
  dflm[i] = le.fit_transform(dflm[i])

# Affiche à nouveau le DataFrame 'dflm' pour visualiser l'impact de l'encodage et vérifier que les colonnes 'Province', 'Saison' et 'ID' contiennent désormais des valeurs numériques
dflm

In [13]:
dflm

,ID,Semaine,Province,Pluviometrie_mm,Temperature_C,NDVI,Humidite_pct,Densite_Population,Couverture_Vaccinale_pct,Tests_Realises,Distance_Centre_Sante_km,Reservoirs_Animaux,Mobilite_Humaine,Cas_Confirmes,Saison,Population_Risque,Taux_Positivite_pct
0,0,1,16,206,24.4,0.480,64,39,31.8,150,22,1,4,29,0,3478,19.33
1,1,2,8,56,23.7,0.474,74,107,24.9,300,5,1,6,42,1,3713,14.00
2,2,3,8,59,25.2,0.333,62,140,53.9,310,16,0,8,20,1,4833,6.45
3,3,4,14,396,24.1,0.354,68,97,25.9,210,4,0,10,82,0,3777,39.05
4,4,5,5,127,19.8,0.303,61,86,30.3,260,24,1,6,24,0,9396,9.23
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2995,2995,2996,9,271,31.6,0.534,88,57,22.4,220,33,1,7,125,0,3166,56.82
2996,2996,2997,21,88,24.6,0.262,95,47,34.1,370,7,1,6,70,1,6099,18.92
2997,2997,2998,0,268,25.6,0.720,58,18,41.9,190,23,0,2,36,0,5475,18.95
2998,2998,2999,3,211,24.9,0.196,77,28,10.6,370,2,0,2,160,0,4813,43.24


# Importation de la classe MinMaxScaler de scikit-learn pour normaliser les données (les ramener dans un intervalle entre 0 et 1)
from sklearn.preprocessing import MinMaxScaler

# Initialisation d'une instance du normalisateur nommée 'minmax'
minmax = MinMaxScaler()

# Ajuste le normalisateur sur les données et transforme le DataFrame 'dflm' ; le résultat est un tableau NumPy (ndarray) où chaque valeur est comprise entre 0 et 1
dflm_min = minmax.fit_transform(dflm)

# Convertit le tableau NumPy normalisé 'dflm_min' en un DataFrame Pandas en conservant les noms des colonnes d'origine
dflm_scaler = pd.DataFrame(dflm_min, columns=dflm.columns)

# Affiche le nouveau DataFrame 'dflm_scaler' pour vérifier visuellement que toutes les données sont bien normalisées entre 0 et 1
dflm_scaler

In [14]:
from sklearn.preprocessing import MinMaxScaler

minmax = MinMaxScaler()
dflm_min = minmax.fit_transform(dflm)

dflm_scaler = pd.DataFrame(dflm_min, columns =dflm.columns)

dflm_scaler

,ID,Semaine,Province,Pluviometrie_mm,Temperature_C,NDVI,Humidite_pct,Densite_Population,Couverture_Vaccinale_pct,Tests_Realises,Distance_Centre_Sante_km,Reservoirs_Animaux,Mobilite_Humaine,Cas_Confirmes,Saison,Population_Risque,Taux_Positivite_pct
0,0.000000,0.000000,0.64,0.512563,0.40000,0.544798,0.436364,0.068687,0.362881,0.084507,0.148936,1.0,0.333333,0.060543,0.0,0.275345,0.203474
1,0.000333,0.000333,0.32,0.135678,0.35625,0.536127,0.618182,0.206061,0.267313,0.295775,0.028369,1.0,0.555556,0.087683,1.0,0.301467,0.147368
2,0.000667,0.000667,0.32,0.143216,0.45000,0.332370,0.400000,0.272727,0.668975,0.309859,0.106383,0.0,0.777778,0.041754,1.0,0.425967,0.067895
3,0.001000,0.001000,0.56,0.989950,0.38125,0.362717,0.509091,0.185859,0.281163,0.169014,0.021277,0.0,1.000000,0.171190,0.0,0.308582,0.411053
4,0.001334,0.001334,0.20,0.314070,0.11250,0.289017,0.381818,0.163636,0.342105,0.239437,0.163121,1.0,0.555556,0.050104,0.0,0.933193,0.097158
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2995,0.998666,0.998666,0.36,0.675879,0.85000,0.622832,0.872727,0.105051,0.232687,0.183099,0.226950,1.0,0.666667,0.260960,0.0,0.240663,0.598105
2996,0.999000,0.999000,0.84,0.216080,0.41250,0.229769,1.000000,0.084848,0.394737,0.394366,0.042553,1.0,0.555556,0.146138,1.0,0.566696,0.199158
2997,0.999333,0.999333,0.00,0.668342,0.47500,0.891618,0.327273,0.026263,0.502770,0.140845,0.156028,0.0,0.111111,0.075157,0.0,0.497332,0.199474
2998,0.999667,0.999667,0.12,0.525126,0.43125,0.134393,0.672727,0.046465,0.069252,0.394366,0.007092,0.0,0.111111,0.334029,0.0,0.423744,0.455158


In [ ]:
# Renvoie et affiche la liste de toutes les colonnes présentes dans le nouveau DataFrame normalisé 'dflm_scaler'
dflm_scaler.columns

In [15]:
dflm_scaler.columns

Index(['ID', 'Semaine', 'Province', 'Pluviometrie_mm', 'Temperature_C', 'NDVI',
       'Humidite_pct', 'Densite_Population', 'Couverture_Vaccinale_pct',
       'Tests_Realises', 'Distance_Centre_Sante_km', 'Reservoirs_Animaux',
       'Mobilite_Humaine', 'Cas_Confirmes', 'Saison', 'Population_Risque',
       'Taux_Positivite_pct'],
      dtype='object')

# Crée la matrice des variables explicatives (X) en supprimant la colonne 'Cas_Confirmes' du DataFrame normalisé le long de l'axe des colonnes (axis=1)
X = dflm_scaler.drop('Cas_Confirmes', axis=1)

# Crée le vecteur de la variable cible (y) en sélectionnant uniquement la colonne 'Cas_Confirmes', qui est la variable que le modèle va tenter de prédire
y = dflm_scaler['Cas_Confirmes']

In [16]:
X = dflm_scaler.drop('Cas_Confirmes', axis=1)
y = dflm_scaler['Cas_Confirmes']

# Affiche la matrice des variables explicatives 'X' pour vérifier sa structure et s'assurer que la colonne cible 'Cas_Confirmes' en a bien été exclue
X

In [17]:
X

,ID,Semaine,Province,Pluviometrie_mm,Temperature_C,NDVI,Humidite_pct,Densite_Population,Couverture_Vaccinale_pct,Tests_Realises,Distance_Centre_Sante_km,Reservoirs_Animaux,Mobilite_Humaine,Saison,Population_Risque,Taux_Positivite_pct
0,0.000000,0.000000,0.64,0.512563,0.40000,0.544798,0.436364,0.068687,0.362881,0.084507,0.148936,1.0,0.333333,0.0,0.275345,0.203474
1,0.000333,0.000333,0.32,0.135678,0.35625,0.536127,0.618182,0.206061,0.267313,0.295775,0.028369,1.0,0.555556,1.0,0.301467,0.147368
2,0.000667,0.000667,0.32,0.143216,0.45000,0.332370,0.400000,0.272727,0.668975,0.309859,0.106383,0.0,0.777778,1.0,0.425967,0.067895
3,0.001000,0.001000,0.56,0.989950,0.38125,0.362717,0.509091,0.185859,0.281163,0.169014,0.021277,0.0,1.000000,0.0,0.308582,0.411053
4,0.001334,0.001334,0.20,0.314070,0.11250,0.289017,0.381818,0.163636,0.342105,0.239437,0.163121,1.0,0.555556,0.0,0.933193,0.097158
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2995,0.998666,0.998666,0.36,0.675879,0.85000,0.622832,0.872727,0.105051,0.232687,0.183099,0.226950,1.0,0.666667,0.0,0.240663,0.598105
2996,0.999000,0.999000,0.84,0.216080,0.41250,0.229769,1.000000,0.084848,0.394737,0.394366,0.042553,1.0,0.555556,1.0,0.566696,0.199158
2997,0.999333,0.999333,0.00,0.668342,0.47500,0.891618,0.327273,0.026263,0.502770,0.140845,0.156028,0.0,0.111111,0.0,0.497332,0.199474
2998,0.999667,0.999667,0.12,0.525126,0.43125,0.134393,0.672727,0.046465,0.069252,0.394366,0.007092,0.0,0.111111,0.0,0.423744,0.455158


# Affiche le vecteur cible 'y' (les valeurs normalisées de 'Cas_Confirmes') pour vérifier sa structure avant de l'injecter dans le modèle
y

In [18]:
y

0       0.060543
1       0.087683
2       0.041754
3       0.171190
4       0.050104
          ...   
2995    0.260960
2996    0.146138
2997    0.075157
2998    0.334029
2999    0.020877
Name: Cas_Confirmes, Length: 3000, dtype: float64

# Importation de la fonction train_test_split de scikit-learn pour diviser le jeu de données en sous-ensembles d'entraînement et de test
from sklearn.model_selection import train_test_split

In [20]:
from sklearn.model_selection import train_test_split

# Divise la matrice des variables explicatives X et le vecteur cible y en sous-ensembles d'entraînement (80%) et de test (20%) en fixant une graine aléatoire (random_state=42) pour garantir la reproductibilité du découpage
x_train, x_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [21]:
x_train, x_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Importation de la classe LinearRegression de la bibliothèque scikit-learn
from sklearn.linear_model import LinearRegression

# Initialisation de l'objet de régression linéaire multiple sous le nom 'model_LRM'
model_LRM = LinearRegression()

# Entraînement (ajustement) du modèle sur les données d'entraînement : l'algorithme calcule les coefficients bêta associés à chaque variable explicative pour prédire au mieux la cible 'y_train'
model_LRM.fit(x_train, y_train)

In [23]:
from sklearn.linear_model import LinearRegression

model_LRM = LinearRegression()

model_LRM.fit(x_train, y_train)

LinearRegression()

# Extrait et affiche l'ordonnée à l'origine (l'intercept bêta_0) du modèle de régression, qui représente la valeur prédite de la variable cible 'Cas_Confirmes' lorsque toutes les variables explicatives sont égales à 0
bo = model_LRM.intercept_
bo

In [25]:
bo = model_LRM.intercept_
bo


np.float64(-0.10849627530053607)

# Interprétation de l'intercept
1. Interprétation de l'Intercept ($\beta_0 = -0.1085$)L'ordonnée à l'origine est négative. Cela signifie que lorsque toutes les variables explicatives de ton modèle sont à leur niveau de référence (égales à 0 après normalisation), la valeur théorique prédite des cas confirmés se situe légèrement en dessous du point minimal de l'échelle. Graphiquement, cela décale la droite de régression vers le bas pour compenser l'effet massif des variables explicatives fortement positives.

# Extrait les coefficients de régression (les poids bêta) associés à chaque variable explicative et les stocke dans la variable 'coef'
coef = model_LRM.coef_

# Affiche le tableau (array) contenant les valeurs numériques de ces coefficients
coef

In [26]:
coef = model_LRM.coef_
coef

array([-9.91877692e-04, -9.91877692e-04, -1.48110084e-03, -1.49967646e-02,
        3.52195286e-03, -3.20427513e-03, -1.00281839e-02, -8.75712209e-03,
        7.42522203e-03,  4.06852275e-01,  1.71146065e-02, -2.36496809e-03,
       -8.54987124e-04,  1.97685499e-05,  9.14417834e-04,  6.10839618e-01])

# Interprétation
Voici les trois axes la base de ces résultats :
1. Les moteurs majeurs : L'effet mécanique du dépistage (+0,6108 et +0,4069)L'analyse montre sans surprise que le Taux_Positivite_pct ($\beta \approx +0,61$) et le nombre de Tests_Realises ($\beta \approx +0,41$) sont les deux variables qui dictent la quasi-totalité de la prédiction du modèle.Ce que cela signifie : Plus on teste et plus la proportion de tests positifs est haute, plus le nombre de cas confirmés enregistrés explose. C'est le reflet d'une épidémie active et d'une surveillance renforcée sur le terrain.
2. Le signal environnemental : Le rôle de la météo et de l'humidité (-0,0150 et -0,0100)Bien que leur poids soit beaucoup plus modeste, la Pluviometrie_mm et l'Humidite_pct affichent les coefficients négatifs les plus nets.Ce que cela signifie : Une augmentation des pluies et de l'humidité globale est associée à une légère diminution du nombre de cas dans le modèle. Cela suggère une dynamique épidémiologique sensible aux variations climatiques saisonnières (par exemple, des pics de transmission potentiellement plus marqués en saison sèche, propices à certains contacts ou comportements en forêt).
3. Les facteurs structurels secondairesDes variables comme le NDVI (indice de végétation), la Densite_Population ou la Distance_Centre_Sante_km ont des coefficients très proches de zéro. Dans le cadre de ce modèle linéaire global, leur impact direct sur la variance est masqué par le poids écrasant des données de dépistage biologique (tests et positivité).

In [28]:
y_pred = model_LRM.predict(x_test)
y_pred

array([ 1.82331703e-01,  7.24608514e-02,  4.76238149e-01,  8.76223715e-02,
        7.13639565e-02,  3.58581199e-01,  2.82518529e-01,  2.58541337e-01,
        9.93192303e-02, -2.66190164e-02,  1.96392426e-01,  1.90572120e-01,
       -4.65848102e-02,  4.06411282e-01,  9.53041580e-02,  2.92459823e-02,
        4.98942084e-01,  2.53364588e-01,  4.76473701e-01,  8.90623862e-02,
        4.90994144e-01, -7.35162390e-03,  1.33140204e-01,  1.86294072e-01,
        3.44378630e-01,  9.87808970e-02,  2.31309514e-01, -1.52195905e-02,
        5.59661299e-02,  5.80692806e-02,  2.48398726e-01,  3.05157186e-02,
       -3.30603915e-02,  1.80871304e-01,  4.21144991e-01,  2.90878684e-01,
        6.94251237e-02,  2.52151390e-01,  3.85168745e-01,  8.40773666e-03,
        2.22366439e-01, -2.79516126e-02,  3.45178386e-02, -2.68795011e-02,
       -3.83843146e-02,  3.95732382e-02,  2.69961238e-01,  6.48438447e-02,
        9.33342548e-02,  3.54108362e-01,  2.18047238e-01,  9.38120008e-02,
        3.30465562e-01,  

# Importation des métriques d'évaluation de la bibliothèque scikit-learn pour mesurer les erreurs et la précision du modèle
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score

# Importation du module standard 'math' pour effectuer des opérations mathématiques (comme la racine carrée pour la RMSE)
import math

In [30]:
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
import math

# Génère les prédictions du modèle sur les données de test (x_test) à l'aide du modèle de régression linéaire entraîné
y_pred = model_LRM.predict(x_test)

# Calcule et affiche l'Erreur Quadratique Moyenne (MSE) en comparant les vraies valeurs (y_test) aux valeurs prédites (y_pred)
print("MSE :", mean_squared_error(y_test, y_pred))

# Calcule et affiche l'Erreur Absolue Moyenne (MAE), qui indique l'écart moyen absolu entre les prédictions et les vraies valeurs
print("MAE :", mean_absolute_error(y_test, y_pred))

# Calcule et affiche la Racine de l'Erreur Quadratique Moyenne (RMSE) pour exprimer l'erreur dans la même unité que les données normalisées
print("RMSE :", math.sqrt(mean_squared_error(y_test, y_pred)))

# Calcule et affiche le Coefficient de Détermination (R²), qui mesure la proportion de la variance de la cible expliquée par le modèle
print("R2 :", r2_score(y_test, y_pred))

In [31]:
print("MSE",mean_squared_error(y_test, y_pred))
print("MAE",mean_absolute_error(y_test, y_pred))
#print("RMSE",math.sqrt(mean_squared_error(y_test, y_pred)))
print("R2", r2_score(y_test, y_pred) )

MSE 0.0015748842824879533
MAE 0.026038586328624155
R2 0.9325384824006935


# Interprétation des Résultats
1. Le Coefficient de Détermination ($R^2 = 0.9325$)Interprétation statistique : Le $R^2$ est proche de 1, ce qui indique une excellente performance globale. Très précisément, $93,25\%$ de la variance (la fluctuation) du nombre de cas confirmés est expliquée par les variables indépendantes incluses dans ton modèle de régression.Signification contextuelle : C'est un score particulièrement élevé pour des données réelles. Cela démontre que les variables explicatives sélectionnées (facteurs spatiaux, temporels, démographiques ou régionaux) capturent de manière très robuste la dynamique et l'intensité des cas.
2. L'Erreur Absolue Moyenne ($\text{MAE} = 0.0260$)Interprétation statistique : La MAE mesure la magnitude moyenne des erreurs de prédiction, sans tenir compte de leur direction.Signification contextuelle : Comme tes données ont été normalisées entre 0 et 1 avec le MinMaxScaler, une MAE de $0,026$ signifie qu'en moyenne, les prédictions du modèle ne s'écartent que de $2,6\%$ de la valeur réelle sur l'échelle standardisée. C'est le signe d'un modèle très précis qui colle de près à la réalité terrain.
3. L'Erreur Quadratique Moyenne ($\text{MSE} = 0.00157$)Interprétation statistique : La MSE mesure la moyenne des carrés des erreurs. Elle accorde un poids prépondérant aux grands écarts.Signification contextuelle : La valeur est extrêmement faible ($0,00157$). Cela t'indique que le modèle commet très peu de "grosses erreurs" ou d'aberrations majeures dans ses prédictions. Les résidus (les écarts entre les cas réels et les cas prédits) sont concentrés très près de zéro.